src/gui/session.py

```diff
@@ -22,5 +24,27 @@ def unload(self):
         del self.action
 
     def run(self):
-        QMessageBox.information(None, "Minimal plugin",
-                                "Do something Fred here")
+        if self.session.IsDataBaseLayerExistsInCurrentProject():
+            QMessageBox.information(
+                None,
+                "Sammo-Boat plugin",
+                "The session datas layer already exists "
+                "in the current project",
+            )
+            return
+
+        workingDirectory = QFileDialog.getExistingDirectory(
+            None, "Select a working directory", QDir.currentPath()
+        )
+        if not workingDirectory:
+            # no directory selected
+            return
+
+        print("Le répertoire sélectionné = " + workingDirectory)
+
+        if not self.session.IsDataBaseAvailableInThisDirectory(
+                workingDirectory):
+            # No geopackage DB in this directory
+            self.session.CreateEmptyDataBase(workingDirectory)
+
+        self.session.LoadDataBase(workingDirectory)
```

Comment:
For now I propose to not load layers by default.

Reply:
Done